src/components/DeleteProbeButton/DeleteProbeButton.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useState } from 'react';
+import { createPortal } from 'react-dom';
+import { Button, Tooltip } from '@grafana/ui';
+import { css } from '@emotion/css';
+
+import type { BackendError, DeleteProbeButtonProps } from './DeleteProbeButton.types';
+import { useDeleteProbe } from 'data/useProbes';
+import { useCanEditProbe } from 'hooks/useCanEditProbe';
+import { ConfirmModal } from 'components/ConfirmModal';
+import { ProbeUsageLink } from 'components/ProbeUsageLink';
+
+import { getPrettyError } from './DeleteProbeButton.utils';
+
+export function DeleteProbeButton({ probe }: DeleteProbeButtonProps) {
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const onDeleteSuccess = useCallback(() => {
+    setShowDeleteModal(false);
+  }, []);
+
+  const { mutateAsync: deleteProbe, isPending } = useDeleteProbe({ onSuccess: onDeleteSuccess });
+  const canEdit = useCanEditProbe(probe);
+  const canDelete = canEdit && !probe.checks.length;
+  const styles = getStyles();
+  const [error, setError] = useState<undefined | { name: string; message: string }>();
+
+  const handleError = (error: Error | BackendError) => {
+    if (!error) {
+      setError(undefined);
+    }
+
+    setError(getPrettyError(error, probe));
+    // if ('data' in error && 'err' in error.data && 'msg' in error.data) {
+    //   setError({ name: error.data.err, message: error.data.msg });
+    // } else {
+    //   setError({ name: 'Unknown error', message: 'An unknown error occurred' });
+    // }
+  };
+
+  const handleOnClick = () => {
+    setShowDeleteModal(true);
+  };
+
+  if (!canDelete) {
+    const tooltipContent = canEdit ? (
+      <>
+        Unable to delete the probe because it is currently in use.
+        <br />
+        <ProbeUsageLink variant="bodySmall" probe={probe} />.
+      </>
+    ) : (
+      <>
+        You do not have sufficient permissions
+        <br />
+        to delete the probe <span className={styles.probeName}>&apos;{probe.name}&apos;</span>.
+      </>
+    );
+
+    return (
+      <Tooltip content={tooltipContent} interactive={canEdit && !canDelete}>
```

Comment:
You can add the `tooltip` prop directly to button now and it works better for accessibility as it becomes keyboard accessible and reads out the tooltip despite the button being 'disabled'.

With that said, I don't love the link in the tooltip as Grafana doesn't support accessing interactive content via the keyboard in tooltips but as there is an alternative presentation of the link in the sidebar this is okay 😄